buildlib/az-helpers.sh

```diff
@@ -247,3 +257,149 @@ get_arch() {
+        echo -e "${COLORS[GREEN]}Container Image:${COLORS[NC]} $(awk -F= '/^PRETTY_NAME=/ {gsub(/"/, "", $2); print $2}' /etc/os-release)"
+        log_docker_variables
+    else
+        echo -e "${COLORS[GREEN]}Running in container:${COLORS[NC]} No"
+        echo -e "${COLORS[GREEN]}Host Machine:${COLORS[NC]} $(uname -n)"
+    fi
+    echo -e "${COLORS[GREEN]}OS:${COLORS[NC]} $(uname -s) ($(uname -o))"
+    echo -e "${COLORS[GREEN]}Kernel:${COLORS[NC]} $(uname -r) ($(uname -v))"
+    echo -e "${COLORS[GREEN]}Architecture:${COLORS[NC]} $(uname -m)"
+}
+
+function log_docker_variables() {
+    if env | grep -q 'DOCKER_OPT_'; then
+        echo -e "${COLORS[GREEN]}Docker Variables:${COLORS[NC]}"
+        for var in DOCKER_OPT_VOLUMES DOCKER_OPT_IB DOCKER_OPT_GPU DOCKER_OPT_ARGS; do
+            [ -n "${!var}" ] && echo -e "    ${!var}"
+        done
+    else
+        echo -e "${COLORS[RED]}    No Docker-specific environment variables set.${COLORS[NC]}"
+    fi
+}
+
+function log_ucx_env_vars() {
+    echo -e "\n${COLORS[CYAN]}UCX Environment Variables${COLORS[NC]}"
+    echo -e "${COLORS[BLUE]}----------------------------${COLORS[NC]}"
+    if env | grep -q '^UCX_'; then
+        env | grep '^UCX_' | sed 's/^/    /'
+    else
+        echo -e "${COLORS[RED]}    No UCX-specific environment variables set.${COLORS[NC]}"
+    fi
+}
+
+function log_ucx_info() {
+    echo -e "\n${COLORS[CYAN]}UCX Information${COLORS[NC]}"
+    echo -e "${COLORS[BLUE]}----------------------------${COLORS[NC]}"
+
+    local command
+    if command -v ucx_info &> /dev/null; then
+        command=ucx_info
+    elif [ -x "$ucx_inst/bin/ucx_info" ]; then
+        command="$ucx_inst/bin/ucx_info"
+    elif [ -x "./src/tools/info/ucx_info" ]; then
+        command="./src/tools/info/ucx_info"
+    else
+        echo -e "${COLORS[RED]}ucx_info command not found.${COLORS[NC]}"
+        return
+    fi
+
+    declare -A section_names=(
+        [-v]="    Version"
+        [-d]="    Devices and Transports"
+        [-b]="    Build Configuration"
+        [-p]="    UCP Context Information"
+        [-w]="    UCP Worker Information"
+        [-e]="    UCP Endpoint Configuration"
+        [-uart]="    UCP Context Features to Use"
+    )
+
+    ordered_flags=(-v -b -d -p -w -e -uart)
+
+    for info_cmd in "${ordered_flags[@]}"; do
+        echo "##[group] ${section_names[$info_cmd]}"
+        if output=$($command $info_cmd 2>&1); then
+            echo "$output" | sed 's/^/    /'
+        else
+            echo -e "${COLORS[RED]}Error: ${output}${COLORS[NC]}"
+        fi
+        echo "##[endgroup]"
+    done
+}
+
+function log_loaded_modules() {
+    echo -e "\n${COLORS[CYAN]}Modules Loaded${COLORS[NC]}"
+    echo -e "${COLORS[BLUE]}----------------------------${COLORS[NC]}"
+    if [ ${#loaded_modules[@]} -eq 0 ]; then
+        echo -e "${COLORS[RED]}    No modules loaded.${COLORS[NC]}"
+    else
+        for module in "${loaded_modules[@]}"; do
```

Comment:
Is that possible to print output of `module list` here instead of reminding loaded modules to special variable?

Reply:
No, because those modules are unloaded before the exit log, which is why I added a variable.
Will add a comment to explain it though.

Reply:
Added a comment